### ISSUE TITLE
All-numeric query prefix causes NullPointerException with SimpleQueryAnalyzer and analyze_wildcard: true

### ISSUE THREAD

**Issue by jamestait** [elastic/elasticsearch#18444]
**Elasticsearch version**: 2.3.2 (also seen in 2.2.0)

**JVM version**: OpenJDK Runtime Environment (IcedTea 2.6.6) (7u101-2.6.6-0ubuntu0.14.04.1)
OpenJDK 64-Bit Server VM (build 24.95-b01, mixed mode)

**OS version**: Ubuntu 14.04.4 LTS

**Description of the problem including expected versus actual behavior**:
When issuing a `simple_query_string` query containing a numeric-prefix term (e.g. "1234*") against a field that uses the `SimpleQueryAnalyzer` using `analyze_wildcard: true`, Elasticsearch throws a `NullPointerException`.

**Steps to reproduce**:

    #!/bin/bash
    
    export ELASTICSEARCH_ENDPOINT="http://localhost:9200"
    
    # Create indexes
    
    curl -XPUT "$ELASTICSEARCH_ENDPOINT/play" -d '{
        "mappings": {
            "type": {
                "properties": {
                    "prop1": {
                        "type": "string",
                        "analyzer": "simple"
                    },
                    "prop2": {
                        "type": "string",
                        "analyzer": "whitespace"
                    }
                }
            }
        }
    }'
    
    # Index documents
    curl -XPOST "$ELASTICSEARCH_ENDPOINT/_bulk?refresh=true" -d '
    {"index":{"_index":"play","_type":"type"}}
    {"prop1":"value1","prop2":"value2"}
    '
    
    # Do searches
    
    curl -XPOST "$ELASTICSEARCH_ENDPOINT/_search?pretty" -d '
    {
        "query": {
            "simple_query_string": {
                "query": "1234*",
                "fields": [
                    "prop1"
                ],
                "analyze_wildcard": true
            }
        }
    }
    '
    
    curl -XPOST "$ELASTICSEARCH_ENDPOINT/_search?pretty" -d '
    {
        "query": {
            "simple_query_string": {
                "query": "1234*",
                "fields": [
                    "prop2"
                ],
                "analyze_wildcard": true
            }
        }
    }
    '

**Provide logs (if relevant)**:
```
[2016-05-18 14:20:53,093][DEBUG][action.search.type       ] [Amphibius] [play2][4], node[HgT_hAH2RC6wTntYZy7rPQ], [P], v[2], s[STARTED], a[id=L9qtfcf9SyyVKb-HZxap9A]: Failed to execute [org.elasticsearch.action.search.SearchRequest@7873a5ad] lastShard [true]
RemoteTransportException[[Amphibius][127.0.0.1:9300][indices:data/read/search[phase/query]]]; nested: SearchParseException[failed to parse search source [
{
    "query": {
        "simple_query_string": {
            "query": "1234*",
            "fields": [
                "prop1"
            ],
            "analyze_wildcard": true
        }
    }
}
]]; nested: NullPointerException;
Caused by: SearchParseException[failed to parse search source [
{
    "query": {
        "simple_query_string": {
            "query": "1234*",
            "fields": [
                "prop1"
            ],
            "analyze_wildcard": true
        }
    }
}
]]; nested: NullPointerException;
        at org.elasticsearch.search.SearchService.parseSource(SearchService.java:853)
        at org.elasticsearch.search.SearchService.createContext(SearchService.java:652)
        at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:618)
        at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:369)
        at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:368)
        at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:365)
        at org.elasticsearch.transport.TransportService$4.doRun(TransportService.java:350)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
        at org.elasticsearch.index.query.SimpleQueryParser.newPrefixQuery(SimpleQueryParser.java:131)
        at org.apache.lucene.queryparser.simple.SimpleQueryParser.consumeToken(SimpleQueryParser.java:406)
        at org.apache.lucene.queryparser.simple.SimpleQueryParser.parseSubQuery(SimpleQueryParser.java:212)
        at org.apache.lucene.queryparser.simple.SimpleQueryParser.parse(SimpleQueryParser.java:152)
        at org.elasticsearch.index.query.SimpleQueryStringParser.parse(SimpleQueryStringParser.java:212)
        at org.elasticsearch.index.query.QueryParseContext.parseInnerQuery(QueryParseContext.java:256)
        at org.elasticsearch.index.query.IndexQueryParserService.innerParse(IndexQueryParserService.java:303)
        at org.elasticsearch.index.query.IndexQueryParserService.parse(IndexQueryParserService.java:206)
        at org.elasticsearch.index.query.IndexQueryParserService.parse(IndexQueryParserService.java:201)
        at org.elasticsearch.search.query.QueryParseElement.parse(QueryParseElement.java:33)
        at org.elasticsearch.search.SearchService.parseSource(SearchService.java:836)
        ... 10 more
```

**Comment by jamestait** [comment]
Also worth noting that floating-point numbers exhibit the same failure.

**Comment by dakrone** [comment]
@jamestait this is actually caused by the same thing as https://github.com/elastic/elasticsearch/issues/18202 and is already fixed, it'll be out in the 2.3.3 release
